run-functional.py

```diff
@@ -26,9 +43,51 @@ def get_tests(test_directory):
 
 
 def launch_ansible_test(test_to_launch, test_directory, test_type, invocation, failure_count):
+
+    inventory = config.get('General', 'inventory', fallback=None) if config else None
+
+    extra_vars_file = config.get('General', 'extra_vars', fallback=None) if config else None
+    extravars = None
+
+    private_data_dir = test_directory + '/' + test_to_launch
+    output_dir = config.get('General', 'output_dir', fallback=None) if config else None
+    if output_dir:
+        private_data_dir = output_dir + '/' + test_to_launch
+        os.makedirs(private_data_dir, mode=0o700, exist_ok=True)
+    else:
+        private_data_dir = test_directory + '/' + test_to_launch
+
+    if config and config.has_section('Ansible Runner Settings'):
+        settings = dict(config.items('Ansible Runner Settings'))
+    else:
+        settings = None
+
+     # ansible_runner.interface.run _SHOULD_ take a dict here. But it doesn't )-:
+     # So instead we'll write a yaml file into the output dir, this seem to work...
+    if settings and output_dir:
+
+        # first convert from strings
+        for v in settings:
+            try:
+                settings[v] = ast.literal_eval(settings[v])
+            except ValueError:
+                pass
+
+        os.makedirs(private_data_dir + '/env', mode=0o700, exist_ok=True)
+        if os.path.exists(private_data_dir + '/env/settings'):
+            os.remove(private_data_dir + '/env/settings')
+        with open(private_data_dir + '/env/settings', 'w') as f:
+            yaml.safe_dump(settings, f)
+
+    if extra_vars_file:
+        with open(extra_vars_file, 'r') as f:
+            extravars = yaml.safe_load(f)
+
     (t, r) = ansible_runner.interface.run_async(
-        private_data_dir=test_directory + '/' + test_to_launch,
-        playbook=test_type + '.yml',
+        private_data_dir=private_data_dir,
+        playbook=test_directory + '/' + test_to_launch + '/' + test_type + '.yml',
```

Comment:
does playbook really need to have test_directory defined since it is defined in private_data_dir?